test/date_range_test.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'test_helper'
+
+describe "Date Range class" do
+  before do
+    @data_range = Hotel::DateRange.new(
+      start_date = Date.new(2010,03,02),
+      end_date = Date.new(2010,03,05)
+    )
+    @date_range_two = Hotel::DateRange.new(
+      start_date = Date.new(2010,03,8),
+      end_date = Date.new(2010,03,10)
+    )
+    @date_range_three = Hotel::DateRange.new(
+      start_date = Date.new(2010,03,8),
+      end_date = Date.new(2010,03,10)
+    )
+  end
+
+  it "is an instance of Date Range" do
+    expect(@data_range).must_be_kind_of Hotel::DateRange
+  end
+
+  describe "overlap?" do
+    it "will return false if the time is not overlapping" do
+      expect(@date_range_two.overlap?(@data_range)).must_equal false
+    end
+
+    it "will return true if the time is overlapping" do
```

Comment:
These are good nominal case tests. It is important to thoroughly test the different ways that dates can overlap. See this pseudo code from the design-scaffold
```
  describe "overlap?" do
    before do
      start_date = Date.new(2017, 01, 01)
      end_date = start_date + 3

      @range = Hotel::DateRange.new(start_date, end_date)
    end

    it "returns true for the same range" do
      start_date = @range.start_date
      end_date = @range.end_date
      test_range = Hotel::DateRange.new(start_date, end_date)

      expect(@range.overlap?(test_range)).must_equal true
    end

    xit "returns true for a contained range" do
    end

    xit "returns true for a range that overlaps in front" do
    end

    xit "returns true for a range that overlaps in the back" do
    end

    xit "returns true for a containing range" do
    end

    xit "returns false for a range starting on the end_date date" do
    end

    xit "returns false for a range ending on the start_date date" do
    end

    xit "returns false for a range completely before" do
    end

    xit "returns false for a date completely after" do
    end
  end

  xdescribe "include?" do
    it "reutrns false if the date is clearly out" do
    end

    it "returns true for dates in the range" do
    end

    it "returns false for the end_date date" do
    end
  end
```